src/geojson_types.jl

```diff
@@ -0,0 +1,350 @@
+    LineString{D,T}(; bbox=nothing, coordinates=nothing) where {D,T} = LineString{D,T}(bbox, coordinates)
+end
+LineString(; bbox=nothing, coordinates::Vector{NTuple{2,T}}) where {T} = LineString{2,T}(D, T, bbox, coordinates)
+Base.eltype(::Type{LineString{D,T}}) where {D,T} = NTuple{D,T}
+
+"""
+    Polygon{D,T}(coordinates::Union{Nothing,Vector{Vector{NTuple{D,T}}}})
+
+A Polygon geometry with `D` dimensions.
+"""
+struct Polygon{D,T} <: AbstractGeometry{D,T}
+    bbox::Union{Nothing,Vector{T}}
+    coordinates::Union{Nothing,Vector{Vector{NTuple{D,T}}}}
+    Polygon{D,T}(bbox, coordinates) where {D,T} = new{D,T}(bbox, coordinates)
+    Polygon{D,T}(; bbox=nothing, coordinates=nothing) where {D,T} = Polygon{D,T}(bbox, coordinates)
+end
+Polygon(; bbox=nothing, coordinates::Vector{Vector{NTuple{2,T}}}) where {T} = Polygon{2,T}(bbox, coordinates)
+Base.eltype(::Type{Polygon{D,T}}) where {D,T} = Vector{NTuple{D,T}}
+
+"""
+    MultiPoint{D,T}(coordinates::Union{Nothing,Vector{NTuple{D,T}}})
+
+A MultiPoint geometry with `D` dimensions.
+"""
+struct MultiPoint{D,T} <: AbstractGeometry{D,T}
+    bbox::Union{Nothing,Vector{T}}
+    coordinates::Union{Nothing,Vector{NTuple{D,T}}}
+    MultiPoint{D,T}(bbox, coordinates) where {D,T} = new{D,T}(bbox, coordinates)
+    MultiPoint{D,T}(; bbox=nothing, coordinates=nothing) where {D,T} = MultiPoint{D,T}(bbox, coordinates)
+end
+MultiPoint(; bbox=nothing, coordinates::Vector{NTuple{2,T}}) where {T} = MultiPoint{2,T}(bbox, coordinates)
+Base.eltype(::Type{MultiPoint{D,T}}) where {D,T} = Vector{NTuple{D,T}}
+
+"""
+    MultiLineString{D,T}(coordinates::Union{Nothing,Vector{Vector{NTuple{D,T}}}})
+
+A MultiLineString geometry with `D` dimensions.
+"""
+struct MultiLineString{D,T} <: AbstractGeometry{D,T}
+    bbox::Union{Nothing,Vector{T}}
+    coordinates::Union{Nothing,Vector{Vector{NTuple{D,T}}}}
+    MultiLineString{D,T}(bbox, coordinates) where {D,T} = new{D,T}(bbox, coordinates)
+    MultiLineString{D,T}(; bbox=nothing, coordinates=nothing) where {D,T} = MultiLineString{D,T}(bbox, coordinates)
+end
+MultiLineString(; bbox=nothing, coordinates::Vector{Vector{NTuple{2,T}}}) where {T} = MultiLineString{2,T}(bbox, coordinates)
+Base.eltype(::Type{MultiLineString{D,T}}) where {D,T} = Vector{NTuple{D,T}}
+
+"""
+    MultiPolygon{D,T}(coordinates::Union{Nothing,Vector{Vector{Vector{NTuple{D,T}}}}})
+
+A MultiPolygon geometry with `D` dimensions.
+"""
+struct MultiPolygon{D,T} <: AbstractGeometry{D,T}
+    bbox::Union{Nothing,Vector{T}}
+    coordinates::Union{Nothing,Vector{Vector{Vector{NTuple{D,T}}}}}
+    MultiPolygon{D,T}(bbox, coordinates) where {D,T} = new{D,T}(bbox, coordinates)
+    MultiPolygon{D,T}(; bbox=nothing, coordinates=nothing) where {D,T} = MultiPolygon{D,T}(bbox, coordinates)
+end
+MultiPolygon(; bbox=nothing, coordinates::Vector{Vector{Vector{NTuple{2,T}}}}) where {T} = MultiPolygon{2,T}(bbox, coordinates)
+Base.eltype(::Type{MultiPolygon{D,T}}) where {D,T} = Vector{Vector{NTuple{D,T}}}
```

Comment:
What is `eltype` for ?

And if you are dispatching on `Vector` anyway why not have another method that catches `D` ?

```julia
MultiPolygon(; bbox=nothing, coordinates::Vector) where {T} = 
    MultiPolygon(bbox, coordinates)

MultiPolygon(bbox, coordinates::Vector{Vecor{NTuple{D,T}) where {D,T} = 
    MultiPolygon(bbox, coordinates::Vector{Vector{NTuple{D,T}}})
```

Then 2d and 3d work.

Reply:
Just updated the example for a typeo, this wont hit unbound args now

Reply:
Hahaha, I wish, but that still throws the test error.

Reply:
Haha wild maybe NTuple includes `D == 0` so there is no T ?? can you Make a union where D is 2, 3, or 4 ??

Reply:
Can I propose we either fix this later, or you try to play around with these types? I'm not even sure how to make D a Union for {2,3,4}.

Reply:
Yeah its annoying with numbers, you need to make a Union over all the types with contained numbers rather than just `{2,3,4}`.

I can PR that - my concern was that it may be a breaking change.

Reply:
Actually lets just merge this, I need to get the other PRs merged too, like writing tables.